core/services/pipeline/task.eth_call.go

```diff
@@ -91,10 +93,16 @@ func (t *ETHCallTask) Run(ctx context.Context, lggr logger.Logger, vars Vars, in
 		return Result{Error: err}, runInfo
 	}
 	var selectedGas uint32
-	if gas > 0 {
-		selectedGas = uint32(gas)
+	if unlimitedGas {
+		if gas > 0 {
+			return Result{Error: errors.New("")}, retryableRunInfo()
```

Comment:
Let's return a useful error here, and should it wrap `ErrBadInput`?
Also, why `retryableRunInfo()` instead of `runInfo`? Wouldn't this be fatal, since it is a configuration of the jobspec?

Reply:
Done.